scaleway/resource_lb_backend_beta.go

```diff
@@ -0,0 +1,301 @@
+package scaleway
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	lb "github.com/scaleway/scaleway-sdk-go/api/lb/v1"
+)
+
+func resourceScalewayLbBackendBeta() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceScalewayLbBackendBetaCreate,
+		Read:   resourceScalewayLbBackendBetaRead,
+		Update: resourceScalewayLbBackendBetaUpdate,
+		Delete: resourceScalewayLbBackendBetaDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"lb_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The load-balancer id",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The name of the backend",
+			},
+			"forward_protocol": {
+				Type: schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					lb.ProtocolTCP.String(),
+					lb.ProtocolHTTP.String(),
+				}, false),
+				Required:    true,
+				Description: "Backend protocol",
+			},
+			"forward_port": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "User sessions will be forwarded to this port of backend servers",
+			},
+			"forward_port_algorithm": {
+				Type: schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					lb.ForwardPortAlgorithmRoundrobin.String(),
+					lb.ForwardPortAlgorithmLeastconn.String(),
+				}, false),
+				Default:     lb.ForwardPortAlgorithmRoundrobin.String(),
+				Optional:    true,
+				Description: "Load balancing algorithm",
+			},
+			"sticky_sessions": {
+				Type: schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{
+					"none",
+					lb.StickySessionsTypeCookie.String(),
+					lb.StickySessionsTypeTable.String(),
+				}, false),
+				Optional:    true,
+				Description: "Load balancing algorithm",
+			},
+			"sticky_sessions_cookie_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
should it be optional computed with a generated name?

Reply:
no this is based on your underlying application for example if you use PHP sticky_sessions_cookie_name will probably be `PHPSESSID` for example